File: backend/src/main/java/com/autocare/mechanic/entity/Mechanic.java
package com.autocare.mechanic.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "mechanics")
public class Mechanic {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private String id;

    // Liga o perfil de mecânico a uma conta de login (users.role = MECHANIC).
    // Sem esse vínculo não há como um usuário autenticado como MECHANIC provar
    // "quem ele é" para o sistema restringir suas ordens de serviço às que lhe
    // foram atribuídas — é o que ServiceOrderService usa para aplicar essa regra.
    // Fica nullable porque nem todo registro de mecânico precisa ter login (pode
    // ser só um cadastro operacional da equipe).
    @Column(name = "user_id", unique = true)
    private String userId;

    @Column(nullable = false)
    private String name;

    private String specialty;

    private String phone;

    @Column(name = "is_available")
    private boolean isAvailable = true;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @Column(name = "is_active")
    private boolean isActive = true;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: backend/app/models/mechanic.py
import uuid

from sqlalchemy import Boolean, ForeignKey, String, Uuid, text
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base
from app.models.mixins import Timestamps, UUIDPrimaryKey


class Mechanic(UUIDPrimaryKey, Timestamps, Base):
    __tablename__ = "mechanics"

    # Liga o perfil de mecânico a uma conta de login (users.role = MECHANIC).
    # Nullable: nem todo mecânico precisa ter login.
    user_id: Mapped[uuid.UUID | None] = mapped_column(
        Uuid, ForeignKey("users.id", ondelete="SET NULL"), unique=True
    )
    name: Mapped[str] = mapped_column(String, nullable=False)
    specialty: Mapped[str | None] = mapped_column(String)
    phone: Mapped[str | None] = mapped_column(String)
    is_available: Mapped[bool] = mapped_column(
        Boolean, default=True, server_default=text("true"), nullable=False
    )
    is_active: Mapped[bool] = mapped_column(
        Boolean, default=True, server_default=text("true"), nullable=False
    )
